source/code/plugins/out_oms_changetracking_file.rb

```diff
@@ -0,0 +1,591 @@
+module Fluent
+
+  class OutChangeTrackingFile < BufferedOutput
+
+    Plugin.register_output('out_oms_oms.changetracking.file', self)
+	
+    # Endpoint URL ex. localhost.local/api/
+
+    def initialize
+      super
+	  
+      require 'base64'
+      require 'digest'
+      require 'json'
+      require 'net/http'
+      require 'net/https'
+      require 'openssl'
+      require 'rexml/document'
+      require 'securerandom'
+      require 'socket'
+      require 'uri'
+      require_relative 'omslog'
+      require_relative 'oms_configuration'
+      require_relative 'oms_common'
+    end
+
+    config_param :omsadmin_conf_path, :string, :default => '/etc/opt/microsoft/omsagent/conf/omsadmin.conf'
+    config_param :cert_path, :string, :default => '/etc/opt/microsoft/omsagent/certs/oms.crt'
+    config_param :key_path, :string, :default => '/etc/opt/microsoft/omsagent/certs/oms.key'
+    config_param :proxy_conf_path, :string, :default => '/etc/opt/microsoft/omsagent/proxy.conf'
```

Comment:
you took my statement "i copied the same content as you have in out_oms_blob and out_oms" out of context and generalized it. 

the code in question, is the lines for "config_param", which are a necessary artifact of fluend plugin. every plugin that expect some parameters have to have them and cannot be avoided. so yes there will be "duplication" of these few lines across all plugins.

Multihoming is all togather a different problem, that has noting to do with fluentd plugins. 
you still havent answered my question? 
"What is the change you require, so this is multihoming compatible if it not currently so"?